File: Python/String-2/end_other.py
# Given two strings, return True if either of the strings appears at the
# very end of the other string, ignoring upper/lower case differences
# (in other words, the computation should not be "case sensitive").
# Note: s.lower() returns the lowercase version of a string.


def end_other(a, b):
  # Make both strings lower case to make the comparison simpler
  a = a.lower()
  b = b.lower()
  
  # Check to see if either string is at the end of the other
  return (a.endswith(b) or b.endswith(a))

File: Java/String-2/endOther.java
/*
 * Given two strings, return true if either of the strings appears at the
 * very end of the other string, ignoring upper/lower case differences
 * (in other words, the computation should not be "case sensitive").
 * Note: str.toLowerCase() returns the lowercase version of a string.
 */

public boolean endOther(String a, String b) {
  // Find which string is shorter to know which string to search for
  int shorter = Math.min(a.length(), b.length());
  
  // Check to see if a string appears at the end of the other
  return (a.substring(a.length() - shorter).equalsIgnoreCase(b) ||
          b.substring(b.length() - shorter).equalsIgnoreCase(a));
}
